extensions/data-plane-selector/data-plane-selector-api/src/main/java/org/eclipse/edc/connector/dataplane/selector/api/v2/validation/DataPlaneInstanceValidator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.selector.api.v2.validation;
+
+import jakarta.json.JsonObject;
+import org.eclipse.edc.validator.jsonobject.JsonObjectValidator;
+import org.eclipse.edc.validator.jsonobject.validators.MandatoryValue;
+import org.eclipse.edc.validator.jsonobject.validators.OptionalIdNotBlank;
+import org.eclipse.edc.validator.spi.Validator;
+
+import static org.eclipse.edc.connector.dataplane.selector.spi.instance.DataPlaneInstance.URL;
+
+/**
+ * Contains the DataPlaneInstance validator definition
+ */
+public class DataPlaneInstanceValidator {
+
+    public static Validator<JsonObject> instance() {
+        return JsonObjectValidator.newValidator()
+                .verifyId(OptionalIdNotBlank::new)
+                .verify(URL, MandatoryValue::new)
```

Comment:
I'd say that also `allowedSourceTypes` and `allowedDestTypes` should be mandatory and they both should have at least an entry

Reply:
Yeah that's correct otherwise it's useless. I Just reported here the same validation on `DataPlaneInstance.Builder`

Reply:
added the min constraint